flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -717,6 +717,11 @@ private static boolean isPointwisePartitioner(StreamPartitioner<?> partitioner)
 	}
 
 	private ResultPartitionType determineResultPartitionType(StreamPartitioner<?> partitioner) {
+
+		if (streamGraph.getCheckpointConfig().isApproximateLocalRecoveryEnabled()) {
+			return ResultPartitionType.PIPELINED_APPROXIMATE;
+		}
```

Comment:
Should this be integrated with the `GlobalDataExchangeMode` and the `shuffleMode` as well? Otherwise we need to document that approximate local recovery disables the global data exchange mode setting and only works if `shuffleMode` is `UNDEFINED`.

Reply:
I've integrated "isApproximateLocalRecoveryEnabled" with GlobalDataExchangeMode in this way:

- GlobalDataExchangeMode is decided based on whether "isApproximateLocalRecoveryEnabled";
- Compatibility with ApproximateLocalRecoveryEnabled is also checked here (unaligned checkpoint can not be used together with approximate local recovery for now).